legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/test/RelationalConnectionFactory.java

```diff
@@ -50,4 +57,25 @@ public Optional<Connection> tryBuildFromConnection(Connection connection, String
     {
         return ConnectionFactoryExtension.super.tryBuildFromConnection(connection, testData, element);
     }
+
+    @Override
+    public Optional<Pair<Connection, List<Closeable>>> tryBuildTestConnection(Connection sourceConnection, EmbeddedData data)
+    {
+        if(data instanceof RelationalData  && sourceConnection instanceof RelationalDatabaseConnection)
```

Comment:
way we designed this, we expect that we will be spinning up H2 server here with data populated here itself and pass that as closable. I didn't see H2 server spawned anywhere, where we spawning that?

Reply:
I think we get this for free when we use the type `LocalH2DatasourceSpecification` for the relational database connection. This is currently how we run the relational mapping tests.